workflow/persist/sqldb/workflowRepository.go

```diff
@@ -0,0 +1,192 @@
+package sqldb
+
+import (
+	"encoding/json"
+	"github.com/argoproj/argo/errors"
+	wfv1 "github.com/argoproj/argo/pkg/apis/workflow/v1alpha1"
+	"github.com/prometheus/common/log"
+	"strings"
+	"time"
+	"upper.io/db.v3/lib/sqlbuilder"
+)
+
+type (
+	WorkflowDBContext struct {
+		TableName            string
+		SupportLargeWorkflow bool
+		Session              sqlbuilder.Database
+	}
+
+	DBRepository interface {
+		Save(wf *wfv1.Workflow) error
+		Get(uid string) *wfv1.Workflow
+		ListAll() []wfv1.Workflow
+		Query(condition interface{}) []wfv1.Workflow
```

Comment:
Get, List, Query need to return errors in the case that we cannot talk to the database.